packages/api/lib/granules.js

```diff
@@ -62,10 +66,48 @@ function getGranuleProductVolume(granuleFiles = []) {
     .reduce((x, y) => x + y, 0);
 }
 
+/**
+ * Delete a Granule from Postgres and Dynamo, delete the Granule's
+ * Files from Postgres and S3
+ *
+ * @param {Object} params
+ * @param {Knex} params.knex - DB client
+ * @param {Object} params.dynamoGranule - Granule from DynamoDB
+ * @param {PostgresGranule} params.pgGranule - Granule from Postgres
+ * @param {FilePgModel} params.filePgModel - File Postgres model
+ * @param {GranulePgModel} params.granulePgModel - Granule Postgres model
+ * @param {Object} params.granuleModelClient - Granule Dynamo model
+ */
+const deleteGranuleAndFiles = async ({
+  knex,
+  dynamoGranule,
+  pgGranule,
+  filePgModel = new FilePgModel(),
+  granulePgModel = new GranulePgModel(),
+  granuleModelClient = new Granule(),
+}) => {
+  const files = await knex(filePgModel.tableName)
```

Comment:
This should be a method on a PG model imo. We could add a basic `query` method to the base model that returns all records matching a WHERE query.

Reply:
Yup, using `knex` directly here should have been a red flag. Here's the new `search` method:

697ef411969ce58029abd0e2c43b6610b8419922